components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java

```diff
@@ -0,0 +1,260 @@
+        metrics = new Metrics(TestUtils.getConfigProvider("metrics.yaml"));
+        metrics.activate();
+        metricService = metrics.getMetricService();
+        metricManagementService = metrics.getMetricManagementService();
+    }
+
+    @BeforeMethod
+    public void init() {
+        metricManagementService.setRootLevel(Level.ALL);
+        metricManagementService.stopReporters();
+        count = 0;
+        eventArrived = false;
+        SiddhiMetricsDataHolder.getInstance().setMetricService(metricService);
+        SiddhiMetricsDataHolder.getInstance().setMetricManagementService(metricManagementService);
+        metricManagementService.startReporter("Console");
+    }
+
+    @AfterSuite
+    protected static void destroy() throws Exception {
+        metrics.deactivate();
+    }
+    @Test
+    public void statisticsMetricsFactory() throws InterruptedException {
+        StatisticsConfiguration statisticsConfiguration = new StatisticsConfiguration(new SiddhiMetricsFactory());
+        SiddhiLatencyMetric latencyTracker= (SiddhiLatencyMetric) statisticsConfiguration
+                .getFactory().createLatencyTracker("test.latency",new SiddhiStatisticsManager(
+                        "MetricsTest",
+                        true));
+        AssertJUnit.assertEquals("test.latency",latencyTracker.getName());
+        SiddhiThroughputMetric throughputTracker = (SiddhiThroughputMetric)statisticsConfiguration
+                .getFactory().createThroughputTracker("test.throughput",new SiddhiStatisticsManager(
+                        "MetricsTest",
+                        true));
+        AssertJUnit.assertEquals("test.throughput",throughputTracker.getName());
+
+        SiddhiMemoryUsageMetric memoryUsageTracker=(SiddhiMemoryUsageMetric)statisticsConfiguration
+                .getFactory()
+                .createMemoryUsageTracker(new SiddhiStatisticsManager("MetricsTest",
+                        true));
+        mockmoryObject mockmoryObject = new mockmoryObject("test.memory");
+        memoryUsageTracker.registerObject(mockmoryObject,"test.memory");
+        AssertJUnit.assertEquals("test.memory",memoryUsageTracker.getName(mockmoryObject));
+        SiddhiBufferedEventsMetric  bufferedEventsTracker=(SiddhiBufferedEventsMetric)statisticsConfiguration
+                .getFactory().createBufferSizeTracker(new SiddhiStatisticsManager("MetricsTest",
+                        true));
+        EventBufferHolder eventBufferHolder = new EventBufferHolder() {
+            @Override
+            public long getBufferedEvents() {
+                return 1;
+            }
+
+            @Override
+            public boolean containsBufferedEvents() {
+                return true;
+            }
+        };
+       bufferedEventsTracker.registerEventBufferHolder(eventBufferHolder,"test.size");
+        AssertJUnit.assertEquals("test.size",bufferedEventsTracker.getName(eventBufferHolder));
+    }
+    @Test
```

Comment:
New line here

Reply:
Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b